### DISCUSSION TITLE
Usage of FIC2 to transfer data from Fabric to MSS - POLARFIRE-SOC

### DISCUSSION THREAD

**Discussion by gsriker77** [polarfire-soc/polarfire-soc-documentation#107] [How to] [answered]
I want to use FIC2 to transfer data from Fabric to MSS. The master of FIC2 can only be the fabric. 
How do I collect data from FIC2 in MSS? Is the destination of FIC2 a buffer/register with the address? 
Is there an example design of using FIC2? The example AXI4_STREAM_DEMO doesn't use FIC2.

Regards,
Sriker.G

**Comment by hughbreslin** [comment, marked as answer]
Hi @gsriker77 🙂 

the FIC2 target interface works in the same way as FIC0 or FIC1, it has access to the coreplex MMIO and the DDR. So any initiator connected to the FIC2 target just needs to address an area in DDR that it would write data to. 

You could modify the AXI stream demo to use FIC 2 by connecting the AXI target interface of the DMA_INITIATOR core to FIC_2_AXI4_TARGET on the MSS and also connecting the S_CLK0 on the DMA_INITIATOR to the FIC_2_CLK output of the CLOCKS_AND_RESETS block. 

You need to make sure you've enabled FIC2's clock and taken it out of reset, this should be done automatically for you by the HSS / MPFS HAL in startup, if you have an older project make sure you include the [mss_config_clk_rst command](https://github.com/polarfire-soc/platform/blob/main/mpfs_hal/startup_gcc/system_startup.c#L181) to take the FIC out of reset and enable its clock.

**Reply by henryding-emcore** [reply]
Hugh, 

Can I use FIC2 without a DDR?

**Reply by hughbreslin** [reply]
Hi @henryding-emcore 

The primary use for FIC2 is accessing the DDR, the [MSS TRM](https://onlinedocs.microchip.com/pr/GUID-0E320577-28E6-4365-9BB8-9E1416A0A6E4-en-US-3/index.html?GUID-98A66493-7C98-436C-9B54-6452BE8BFED2) has a bit more detail on the use cases for the FICs :)

**Reply by henryding-emcore** [reply]
Thanks, Hugh!
